### PR TITLE
[Storage] Deprecate `BlobProperties` as `blob` or `name` parameter input

### PR DESCRIPTION
**Pull request by vincenttran-msft**
As title states 😄

### CODE REVIEW

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py [hunk 1]
```diff
@@ -43,7 +43,6 @@
     from ._lease import BlobLeaseClient
     from ._models import (
         ContainerProperties,
-        BlobProperties,
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py:46]: Point for whole review, did we decided we wanted to add a deprecation warning for people using `BlobProperties`?

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py [hunk 2]
```diff
@@ -566,11 +565,10 @@ def create_container(
...
     @distributed_trace
     def delete_container(
-            self, container,  # type: Union[ContainerProperties, str]
-            lease=None,  # type: Optional[Union[BlobLeaseClient, str]]
-            **kwargs
-        ):
-        # type: (...) -> None
+            self, container: Union["ContainerProperties", str],
+            lease: Optional[Union["BlobLeaseClient", str]] = None,
+            **kwargs: Any
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py:570]: nit: extra tabs here

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py [hunk 3]
```diff
@@ -566,11 +565,10 @@ def create_container(
 
     @distributed_trace
     def delete_container(
-            self, container,  # type: Union[ContainerProperties, str]
-            lease=None,  # type: Optional[Union[BlobLeaseClient, str]]
-            **kwargs
-        ):
-        # type: (...) -> None
+            self, container: Union["ContainerProperties", str],
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py:568]: Let's open a GitHub issue to consider deprecating `ContainerProperties` as well as any other Properties input we have across all packages. I don't think any have a glaring issue like `BlobProperties` does but may be good to be consistent.

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py (hunk 1 above)
Comment (vincenttran-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py:46]: Sure, I think that would be a good idea! This is what Anna suggested on the previous iteration of this PR:
```
import warnings
warnings.warn(
    "The use of a 'BlobProperties' instance is deprecated. Please use str instead.",
    DeprecationWarning
)
```

File: sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py [hunk 4]
```diff
@@ -773,6 +768,11 @@ def get_blob_client(
                 :dedent: 12
                 :caption: Getting the blob client to interact with a specific blob.
         """
+        if isinstance(blob, BlobProperties):
+            warnings.warn(
+                "The use of a 'BlobProperties' instance is deprecated. Please use str instead.",
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-blob/azure/storage/blob/_blob_service_client.py:773]: Can we add a mention of the param `blob` here (and everywhere else). 

"The use of a 'BlobProperties' instance for param blob is deprecated. Please use str instead."

**Review by jalauzon-msft — APPROVED** [review]

**Pull request merged**